packages/daemon/src/db/index.ts

```diff
@@ -426,37 +415,52 @@ export const voidTransaction = async (
-                                   locked_balance = locked_balance - ?,
-                                   transactions = transactions - 1,
-                                   timelock_expires = CASE
-                                                        WHEN timelock_expires IS NULL THEN VALUES(timelock_expires)
-                                                        WHEN VALUES(timelock_expires) IS NULL THEN timelock_expires
-                                                        ELSE LEAST(timelock_expires, VALUES(timelock_expires))
-                                                      END,
-                                   unlocked_authorities = (unlocked_authorities | VALUES(unlocked_authorities)),
-                                   locked_authorities = locked_authorities | VALUES(locked_authorities)`,
-        [
-          entry,
-          tokenBalance.totalAmountSent,  // For total_received subtraction
-          tokenBalance.unlockedAmount,   // For unlocked_balance subtraction
-          tokenBalance.lockedAmount,     // For locked_balance subtraction
-          address,
-          token
-        ],
+      // Check if address_balance entry exists first
+      const [existingRows] = await mysql.query(
+        'SELECT total_received FROM address_balance WHERE address = ? AND token_id = ?',
+        [address, token]
       );
 
+      if (existingRows.length > 0) {
+        // Entry exists, perform UPDATE to subtract values
+        await mysql.query(
+          `UPDATE address_balance
+           SET total_received = total_received - ?,
+               unlocked_balance = unlocked_balance - ?,
+               locked_balance = locked_balance - ?,
+               transactions = transactions - 1,
+               timelock_expires = CASE
+                                    WHEN timelock_expires IS NULL THEN ?
+                                    WHEN ? IS NULL THEN timelock_expires
+                                    ELSE LEAST(timelock_expires, ?)
+                                  END,
+               unlocked_authorities = (unlocked_authorities | ?),
+               locked_authorities = locked_authorities | ?
+           WHERE address = ? AND token_id = ?`,
+          [
+            tokenBalance.totalAmountSent,
+            tokenBalance.unlockedAmount,
+            tokenBalance.lockedAmount,
+            tokenBalance.lockExpires,
+            tokenBalance.lockExpires,
+            tokenBalance.lockExpires,
+            tokenBalance.unlockedAuthorities.toUnsignedInteger(),
+            tokenBalance.lockedAuthorities.toUnsignedInteger(),
+            address,
+            token
+          ]
+        );
+      } else {
+        // Entry doesn't exist, this means the balance was never added in the first place
+        // This can happen in tests that don't simulate the complete transaction flow
+        console.log(`Warning: Trying to void transaction for address ${address} token ${token} but no balance entry exists`);
```

Comment:
suggestion(non-blocking): Change `log` to `warn`
```suggestion
        console.warn(`Warning: Trying to void transaction for address ${address} token ${token} but no balance entry exists`);
```

Reply:
Done